apps/pay/app/[username]/layout.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react"
+
+import UsernameLayoutContainer from "@/components/Layouts/UsernameLayout"
```

Comment:
I don't really think this is useful to have this file here invoking a component. we could have in place the component it, instead of wrapping it in a call to `component`. I generally see component as something that we should do when it is something that can be invoked multiple time, but here this layout is called only once, and I don't think it will be present elsewhere.

removing the layer of indrection when it feels clear it will never be needed in the future make it easier to parse the code

Reply:
`UsernameLayoutContainer` is a client component, if we use it in place then the whole child component will become client by default. Currently, I know most of them are client component, but we are going to do refactoring in UI and remove elements from this, then we can use server component there.

https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props

Reply:
right ok, I forget the nuances of server/client component embedding